src/components/AppModal.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+  <div class="container" v-if="isModalActive">
+    <div class="header">
+      <h2>{{ modalTitle }}</h2>
+      <button @click="toggleAppModal">
+        <i class="fa-solid fa-x"></i>
+      </button>
+    </div>
+    <div class="form_body">
+      <slot />
+    </div>
+    <div class="footer">
+      <button class="cancel" @click="toggleAppModal">Cancel</button>
+      <button class="save">Save</button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapState, mapMutations } from 'vuex';
+
+export default {
+  name: 'AppModal',
+  props: {
+    modalTitle: String,
+  },
+  computed: {
+    ...mapState({
+      isModalActive: 'isModalActive',
+    }),
+  },
+  methods: {
+    ...mapMutations({ toggleAppModal: 'TOGGLE_APP_MODAL' }),
```

Comment:
```suggestion
    ...mapMutations({ toggleAppModal: TOGGLE_APP_MODAL }),
```
This should be the exported constant, think about what would happen if you changed the constant name in the store